src/qs8-vadd/gen/qs8-vadd-minmax-hvx-u128.c

```diff
@@ -0,0 +1,141 @@
+    HVX_VectorPair vb0_i16 = Q6_Wh_vunpack_Vb(vb0);
+    HVX_Vector va0_lo = Q6_V_lo_W(va0_i16);
+    HVX_Vector va0_hi = Q6_V_hi_W(va0_i16);
+    HVX_Vector vb0_lo = Q6_V_lo_W(vb0_i16);
+    HVX_Vector vb0_hi = Q6_V_hi_W(vb0_i16);
+
+    // vacc = vbias + va * va_multiplier + vb * vb_multiplier with widening 16-bit to 32-bit
+    HVX_Vector va0_lo_mul_even = vbias;
+    va0_lo_mul_even = Q6_Vw_vmpyieacc_VwVwVh(va0_lo_mul_even, va_multiplier, va0_lo);
+    HVX_Vector va0_lo_mul_odd = Q6_Vw_vadd_VwVw(vbias, Q6_Vw_vmpyio_VwVh(va_multiplier, va0_lo));
+
+    HVX_Vector vb0_lo_mul_even = Q6_Vw_vmpyie_VwVh(vb_multiplier_lo, vb_multiplier_hi, vb0_lo);
+    HVX_Vector vb0_lo_mul_odd = Q6_Vw_vmpyio_VwVh(vb_multiplier, vb0_lo);
+
+    HVX_Vector vacc0_lo_even = Q6_Vw_vadd_VwVw(va0_lo_mul_even, vb0_lo_mul_even);
+    HVX_Vector vacc0_lo_odd = Q6_Vw_vadd_VwVw(va0_lo_mul_odd, vb0_lo_mul_odd);
+
+    HVX_Vector va0_hi_mul_even = vbias;
+    va0_hi_mul_even = Q6_Vw_vmpyieacc_VwVwVh(va0_hi_mul_even, va_multiplier, va0_hi);
+    HVX_Vector va0_hi_mul_odd = Q6_Vw_vadd_VwVw(vbias, Q6_Vw_vmpyio_VwVh(va_multiplier, va0_hi));
+
+    HVX_Vector vb0_hi_mul_even = Q6_Vw_vmpyie_VwVh(vb_multiplier_lo, vb_multiplier_hi, vb0_hi);
+    HVX_Vector vb0_hi_mul_odd = Q6_Vw_vmpyio_VwVh(vb_multiplier, vb0_hi);
+
+    HVX_Vector vacc0_hi_even = Q6_Vw_vadd_VwVw(va0_hi_mul_even, vb0_hi_mul_even);
+    HVX_Vector vacc0_hi_odd = Q6_Vw_vadd_VwVw(va0_hi_mul_odd, vb0_hi_mul_odd);
+
+    // narrow shift to 16-bit
+    // vacc = vacc + voutput_zero_point
+    HVX_Vector vacc0_lo = Q6_Vh_vasr_VwVwR_sat(vacc0_lo_odd, vacc0_lo_even, first_shift);
+    vacc0_lo = Q6_Vh_vadd_VhVh(voutput_zero_point, Q6_Vh_vasr_VhR(vacc0_lo, rest_shift));
+    HVX_Vector vacc0_hi = Q6_Vh_vasr_VwVwR_sat(vacc0_hi_odd, vacc0_hi_even, first_shift);
+    vacc0_hi = Q6_Vh_vadd_VhVh(voutput_zero_point, Q6_Vh_vasr_VhR(vacc0_hi, rest_shift));
+
+    // narrow 16-bit to 8-bit
+    HVX_Vector vout0 = Q6_Vb_vpack_VhVh_sat(vacc0_hi, vacc0_lo);
+
+    // minmax
+    vout0 = Q6_Vb_vmax_VbVb(voutput_min, vout0);
+    vout0 = Q6_Vb_vmin_VbVb(voutput_max, vout0);
+
+    // store output
+    *((HVX_UVector *) ptr_o) = vout0;
+    ptr_o += 128;
+  }
+  if XNN_UNLIKELY(batch != 0){
+    do {
+      HVX_Vector va = *((HVX_UVector*)input_a);
+      HVX_Vector vb = *((HVX_UVector*)input_b);
+      if XNN_LIKELY(batch > (32 * sizeof(int8_t))) {
+        input_a += 32;
+        input_b += 32;
+      }
+
+      HVX_VectorPair va_i16 = Q6_Wh_vunpack_Vb(va);
+      HVX_Vector va_lo = Q6_V_lo_W(va_i16);
+
+      HVX_VectorPair vb_i16 = Q6_Wh_vunpack_Vb(vb);
+      HVX_Vector vb_lo = Q6_V_lo_W(vb_i16);
+
+      HVX_Vector va_lo_mul_even = vbias;
+      va_lo_mul_even = Q6_Vw_vmpyieacc_VwVwVh(va_lo_mul_even, va_multiplier, va_lo);
+      HVX_Vector va_lo_mul_odd = Q6_Vw_vadd_VwVw(vbias, Q6_Vw_vmpyio_VwVh(va_multiplier, va_lo));
+
+      HVX_Vector vb_lo_mul_even = Q6_Vw_vmpyie_VwVh(vb_multiplier_lo, vb_multiplier_hi, vb_lo);
```

Comment:
Can you use vmpyieacc here too? I think maybe this whole loop body can be:

```
      HVX_Vector vacc_even = vbias;
      vacc_even = Q6_Vw_vmpyieacc_VwVwVh(vacc_even , va_multiplier, va_lo);
      HVX_Vector vacc_odd = Q6_Vw_vadd_VwVw(vbias, Q6_Vw_vmpyio_VwVh(va_multiplier, va_lo));

      vacc_even = Q6_Vw_vmpyieacc_VwVh(vacc_even, vb_multiplier, vb_lo);
      vacc_odd = Q6_Vw_vadd_VwVw(vacc_odd, Q6_Vw_vmpyio_VwVh(vb_multiplier, vb_lo));
```

Reply:
This is much more efficient than the original method. Thank you!